libminifi/src/ResourceClaim.cpp

```diff
@@ -41,11 +42,15 @@ void setDefaultDirectory(std::string path) {
   default_directory_path = path;
 }
 
-ResourceClaim::ResourceClaim(std::shared_ptr<core::StreamManager<ResourceClaim>> claim_manager)
```

Comment:
These claims should not be persisted on stops. They should complete, so making them weak seems counterintuitive, though I admit it may be necessary -- why is the claim required to be weak ?